janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -1582,4 +1583,27 @@ public final boolean isClosed() {
     public boolean hasModifications() {
         return !addedRelations.isEmpty() || !deletedRelations.isEmpty();
     }
+
+    @Override
+    public void expireSchemaElement(final long id) {
+        if (vertexCache.contains(id)) {
+            final InternalVertex v = vertexCache.get(id, externalVertexRetriever);
+            if (v instanceof JanusGraphSchemaVertex) {
+                JanusGraphSchemaVertex sv = (JanusGraphSchemaVertex) v;
+                sv.resetCache();
+                if (sv.getDefinition().containsKey(TypeDefinitionCategory.INTERNAL_INDEX)
+                    && !sv.getDefinition().<Boolean>getValue(TypeDefinitionCategory.INTERNAL_INDEX)) {
+                    try {
+                        // Invalidate mixed index
+                        String store = sv.getDefinition()
+                            .getValue(TypeDefinitionCategory.BACKING_INDEX, String.class);
+                        IndexTransaction indexTx = txHandle.getIndexTransaction(store);
+                        indexTx.invalidate(sv.name());
```

Comment:
I would generally prefer using if else conditions when possible but if that's complicated I think it's fine to use try catch

Reply:
It was added to handle a possible NPE thrown by getIndexTransaction. Instead, I've added a method hasIndexTransaction to avoid the NPE.